core/src/com/unciv/logic/automation/BattleHelper.kt

```diff
@@ -47,6 +47,8 @@ object BattleHelper {
                         // DO NOT use "!unit.movement.canPassThrough(it)" since then we won't be able to
                         // capture enemy units since we can't move through them!
                         && !it.canCivPassThrough(unit.civInfo)
+                        // Land Unit can't capture Naval and vice versa
+                        && !(unit.type.isLandUnit() && it.isWater || unit.type.isWaterUnit() && it.isLand)
```

Comment:
I may be mistaken, but isn't this line in the function for getting the attackable enemies? And so doens't this also stop melee attacks into/from the water?

Reply:
No, since this is in the "mapCombatant.isCivilian" filterNot check

Reply:
Ah, my bad